src/Components/MassageList.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState, useRef } from "react";
+
+export const MassageList = () => {
+ 
+
+
+
+  const [massageList, setMassage] = useState([
+    { author: "Поле сообщений:", massage: "" },
+  ]);
+  const [text, setText] = useState("");
+
+  const pushMassage = () => {
+    setMassage((massageList) => [
+      ...massageList,
+      { author: "User", massage: text },
+    ]);
+    setText("");
+  };
+
+  let ref = useRef(null);
+
+  useEffect(() => {
+    if (massageList.length > 1) {
+      const UserLength = massageList[massageList.length - 1];
+      if (UserLength.author === "User") {
+        console.log(UserLength);
+        clearTimeout(ref.current);
```

Comment:
https://stackoverflow.com/questions/53090432/react-hooks-right-way-to-clear-timeouts-and-intervals